components/shared/net/lib.rs

```diff
@@ -594,54 +595,57 @@ impl ResourceFetchTiming {
+            ResourceAttribute::DomainLookupStart => self.domain_lookup_start = now,
             ResourceAttribute::RedirectCount(count) => self.redirect_count = count,
-            ResourceAttribute::RequestStart => self.request_start = precise_time_ns(),
-            ResourceAttribute::ResponseStart => self.response_start = precise_time_ns(),
+            ResourceAttribute::RequestStart => self.request_start = now,
+            ResourceAttribute::ResponseStart => self.response_start = now,
             ResourceAttribute::RedirectStart(val) => match val {
-                RedirectStartValue::Zero => self.redirect_start = 0,
+                RedirectStartValue::Zero => self.redirect_start = Duration::ZERO,
                 RedirectStartValue::FetchStart => {
-                    if self.redirect_start == 0 {
+                    if self.redirect_start.is_zero() {
                         self.redirect_start = self.fetch_start
                     }
                 },
             },
             ResourceAttribute::RedirectEnd(val) => match val {
-                RedirectEndValue::Zero => self.redirect_end = 0,
+                RedirectEndValue::Zero => self.redirect_end = Duration::ZERO,
                 RedirectEndValue::ResponseEnd => self.redirect_end = self.response_end,
             },
-            ResourceAttribute::FetchStart => self.fetch_start = precise_time_ns(),
+            ResourceAttribute::FetchStart => self.fetch_start = now,
             ResourceAttribute::ConnectStart(val) => self.connect_start = val,
             ResourceAttribute::ConnectEnd(val) => self.connect_end = val,
-            ResourceAttribute::SecureConnectionStart => {
-                self.secure_connection_start = precise_time_ns()
-            },
-            ResourceAttribute::ResponseEnd => self.response_end = precise_time_ns(),
+            ResourceAttribute::SecureConnectionStart => self.secure_connection_start = now,
+            ResourceAttribute::ResponseEnd => self.response_end = now,
             ResourceAttribute::StartTime(val) => match val {
                 ResourceTimeValue::RedirectStart
-                    if self.redirect_start == 0 || !self.timing_check_passed => {},
+                    if self.redirect_start.is_zero() || !self.timing_check_passed => {},
                 _ => self.start_time = self.get_time_value(val),
             },
         }
     }
 
-    fn get_time_value(&self, time: ResourceTimeValue) -> u64 {
+    fn get_time_value(&self, time: ResourceTimeValue) -> Duration {
         match time {
-            ResourceTimeValue::Zero => 0,
-            ResourceTimeValue::Now => precise_time_ns(),
+            ResourceTimeValue::Zero => Duration::ZERO,
+            ResourceTimeValue::Now => SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap_or_default(),
             ResourceTimeValue::FetchStart => self.fetch_start,
             ResourceTimeValue::RedirectStart => self.redirect_start,
```

Comment:
It might make more sense to have this return a `Instant` which is guaranteed to be monotonic.